packages/zmarkdown/client/index.js

```diff
@@ -0,0 +1,51 @@
+export const plugins = {}
+export let defaultType
+
+export function use (obj) {
+  if (!obj) {
+    throw new Error('This is not an object')
+  }
+
+  // Check the object structure
+  if (!('name' in obj)) {
+    throw new Error("missing 'name' in plugin")
+  } else if (typeof obj.name !== 'string') {
+    throw new Error('Plugin name should be a string')
+  }
+  if (!('render' in obj)) {
+    throw new Error("missing 'render' function in plugin")
+  } else if (typeof obj.render !== 'function') {
+    throw new Error('render is not a function')
+  }
+
+  plugins[obj.name] = obj
+}
+
+export function setDefaultType (type) {
+  if (type in plugins) {
+    defaultType = type
+  } else {
+    throw new Error(`Unknown type: ${type}`)
+  }
+}
+
+export function resetDefaultType () {
+  defaultType = null
+}
+
+export function render (str, type = null, cb = null) {
```

Comment:
Please make sure `cb` is used when passed as 2nd arguments and type isn't specified